### ISSUE TITLE
Test failures on master

### ISSUE THREAD

**Issue by amueller** [automl/SMAC3#294]
I get many test failures of this form:
```
======================================================================
ERROR: test_predict (test_epm.test_rfr_with_instances.TestRFWithInstances)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/andy/checkout/SMAC3/test/test_epm/test_rfr_with_instances.py", line 38, in test_predict
    list(map(lambda x: (0, 10), range(10))), dtype=object))
  File "/home/andy/checkout/SMAC3/smac/epm/rf_with_instances.py", line 77, in __init__
    self.rf_opts.seed = seed
  File "/home/andy/anaconda3/lib/python3.6/site-packages/pyrfr/regression.py", line 98, in set_attr
    raise AttributeError("You cannot add attributes to %s" % self)
AttributeError: You cannot add attributes to <pyrfr.regression.forest_opts; proxy of <Swig Object of type 'rfr::forests::forest_options< num_t,response_t,index_t > *' at 0x7ff35a5059c0> >
```
I guess there's something wrong with my pyrfr installation?

**Comment by amueller** [comment]
I installed from the master branch. It looks like ``pyrfr`` is pip-installed, but the package seems to be missing something?

**Comment by amueller** [comment]
Looking into http://automl.github.io/auto-sklearn/stable/installation.html#installing-auto-sklearn right now...

But apparently pyrfr failed to install, but though pip said it succeeded:
```
  TEST FAILED: build/bdist.linux-x86_64/wheel/ does NOT support .pth files
  error: bad install directory or PYTHONPATH
  
  You are attempting to install a package to a directory that is not
  on PYTHONPATH and which Python does not read ".pth" files from.  The
  installation directory you specified (via --install-dir, --prefix, or
  the distutils default setting) was:
  
      build/bdist.linux-x86_64/wheel/
  
  and your PYTHONPATH environment variable currently contains:
  
      ''
  
  Here are some of your options for correcting the problem:
  
  * You can choose a different installation directory, i.e., one that is
    on PYTHONPATH or supports .pth files
  
  * You can add the installation directory to the PYTHONPATH environment
    variable.  (It must then also be on PYTHONPATH whenever you run
    Python and want to use the package(s) you are installing.)
  
  * You can set up the installation directory to support ".pth" files by
    using one of the approaches described here:
  
    https://setuptools.readthedocs.io/en/latest/easy_install.html#custom-installation-locations
  
  
  Please make the appropriate changes for your system and try again.
  
  ----------------------------------------
  Failed building wheel for pyrfr
```

**Comment by amueller** [comment]
I tried to clone the pyrfr repo but the repo mentioned in the docs here doesn't seem to be public any more:
http://pyrfr.readthedocs.io/en/latest/installation.html#installation
```
$ git clone git@bitbucket.org:aadfreiburg/random_forest_run.git
```
>Permission denied (publickey).

That readme also says pyrfr is not on pipy which is not true.

**Comment by amueller** [comment]
I downloaded the .tar.gz from pypi and run ``python setup.py install`` that seemed to have worked.

**Comment by amueller** [comment]
(installing the conda gcc seems to require me recompiling scikit-learn, though).

**Comment by mlindauer** [comment]
Hi Andreas,

let me try to answer hopefully all of your issues:

> `File "/home/andy/anaconda3/lib/python3.6/site-packages/pyrfr/regression.py", line 98, in set_attr
    raise AttributeError("You cannot add attributes to %s" % self)
AttributeError: You cannot add attributes to <pyrfr.regression.forest_opts; proxy of <Swig Object of type 'rfr::forests::forest_options< num_t,response_t,index_t > *' at 0x7ff35a5059c0> >`

Right now, the master branch of SMAC and the master branch of pyrfr are not compatible. There is bug in SMAC which will be fixed in the next SMAC release (hopefully within the next days).
Using the 0.4.0 pyrfr version on pypi should solve the issue for now.

>   Failed building wheel for pyrfr

Unfortunately, the wheel building still fails which gives you this confusing warning. However, this should not really matter in practice.

> http://pyrfr.readthedocs.io/en/latest/installation.html#installation 

Thanks for pointing this out. Indeed, we forgot to update the online documentation of the pyrfr; we will do this very soon.

**Comment by amueller** [comment]
That the wheel building fails means that ``pip install pyrfr`` fails (at
least for me), so that does matter.

Sent from phone. Please excuse spelling and brevity.

On Aug 3, 2017 02:36, "Marius Lindauer" <notifications@github.com> wrote:

Hi Andreas,

let me try to answer hopefully all of your issues:

File "/home/andy/anaconda3/lib/python3.6/site-packages/pyrfr/regression.py",
line 98, in set_attr raise AttributeError("You cannot add attributes to %s"
% self) AttributeError: You cannot add attributes to
<pyrfr.regression.forest_opts; proxy of <Swig Object of type
'rfr::forests::forest_options< num_t,response_t,index_t > *' at
0x7ff35a5059c0> >

Right now, the master branch of SMAC and the master branch of pyrfr are not
compatible. There is bug in SMAC which will be fixed in the next SMAC
release (hopefully within the next days).
Using the 0.4.0 pyrfr version on pypi should solve the issue for now.

Failed building wheel for pyrfr

Unfortunately, the wheel building still fails which gives you this
confusing warning. However, this should not really matter in practice.

http://pyrfr.readthedocs.io/en/latest/installation.html#installation

Thanks for pointing this out. Indeed, we forgot to update the online
documentation of the pyrfr; we will do this very soon.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/automl/SMAC3/issues/294#issuecomment-319881462>, or mute
the thread

**Comment by mlindauer** [comment]
> That the wheel building fails means that ``pip install pyrfr`` fails (at
least for me), so that does matter.

Sorry, could you please elaborate a bit more?
On our systems, we get the same warning ("Failed building wheel for pyrfr"),
but nevertheless, pyrfr is installed correctly and can be used without any issues.

**Comment by amueller** [comment]
Hm, it looks like you're right, and it works with a pip install. I'm a bit confused because I never tried to install  the dev version as far as I remember, so I'm not sure why it failed before. So looks like all is good?

**Comment by mlindauer** [comment]
> So looks like all is good?

if you say so.
Can we close this issue?

**Comment by GillesMoyse** [comment]
Hello, I'm experiencing exactly the same issue than you @amueller, can you please explain what you did in order to fix it? Simply install pyrfr 0.4.0 ?
